src/uri/include/sourcemeta/jsontoolkit/uri.h

```diff
@@ -157,6 +157,19 @@ class SOURCEMETA_JSONTOOLKIT_URI_EXPORT URI {
   /// ```
   [[nodiscard]] auto path() const -> std::optional<std::string>;
 
+  /// Set the path part of the URI. Foe example:
+  ///
+  /// ```cpp
+  /// #include <sourcemeta/jsontoolkit/uri.h>
+  /// @include <cassert>
+  ///
+  /// sourcemeta::jsontoolkit::URI uri{"https://www.sourcemeta.com"};
+  /// uri.path("/foo/bar");
+  /// assert(uri.path().has_value());
+  /// assert(uri.path().value() == "/foo/bar");
+  /// ```
+  auto path(std::string_view path) -> URI &;
```

Comment:
Given we store the data as an `std::string`, I think you might be better off taking `const std::string &`

Reply:
On this, I think you should also have an overload for `std::string &&` for performance reasons

Reply:
> Given we store the data as an `std::string,` I think you might be better off taking `const std::string &`

I have two concerns with that:
- I am not sure to understand the link that there is between how we store data and the interface itself.
- But using `const` I cannot modify `path` and I have to create copy, then I was wondering about potential perf implication. Does that make sense?

Reply:
> On this, I think you should also have an overload for std::string && for performance reasons

I struggled with that haha!

Let me do it in another PR, when I can focus 100% on this.

Reply:
I think on your points above, that is the reason C++ setters usually have both copy and move overloads. Then the developer can pick the one that works best on their specific circumstance!